vehicles.lua

```diff
@@ -1562,9 +1565,24 @@ end
 -- Get the turning radius of the vehicle and its implements (copied from AIDriveStrategyStraight.updateTurnData())
 function AIDriverUtil.getTurningRadius(vehicle)
 	courseplay.debugVehicle(courseplay.DBG_IMPLEMENTS, vehicle, 'Finding turn radius:')
-
-	local radius = vehicle.maxTurningRadius * 1.05 -- TODO: do we really need this buffer?
-	courseplay.debugVehicle(courseplay.DBG_IMPLEMENTS, vehicle, '  maxTurningRadius by Giants is %.1f', vehicle.maxTurningRadius)
+	
+	local radiusMultiplier = 1.05
+	local radius = 5
+	-- default value if all others fail
+	courseplay.debugVehicle(courseplay.DBG_IMPLEMENTS, vehicle, '  turnRadius set to a default %.1f', radius)
+	local TR = ceil(courseplay:calculateTurnRadius(steeringType, wheelBase, rotMax, CPRatio) * radiusMultiplier)
+	-- copied from courseplay:getVehicleTurnRadius
+	if vehicle.maxTurningRadius then
+			radius = vehicle.maxTurningRadius * radiusMultiplier
+			courseplay.debugVehicle(courseplay.DBG_IMPLEMENTS, vehicle, '  maxTurningRadius by Giants is %.1f', radius)
+	elseif g_vehicleConfigurations:get(vehicle, 'turnRadius') then
```

Comment:
I believe this is the only thing we need to add here, we were checking if there's a configured turn radius for any of the implements, but did not check the vehicle itself.

BTW, this is the place where we have to address #6944 as well, checking for any user set radius.

Reply:
left this and added an additional if-statement to check for a user set radius (at least I hope so)